### ISSUE TITLE
Tests failed :: TypeError: calling <class 'BayesicFitting.source.kernels.Uniform.Uniform'> 

### ISSUE THREAD

**Issue by Hellseher** [dokester/BayesicFitting#19]
Hi,

During packing your project for Guix I faced with this issue in tests:

```
Traceback (most recent call last):
  File "/tmp/guix-build-python-bayesicfitting-3.0.1.drv-0/BayesicFitting-3.0.1/setup.py", line 9, in <module>
    setup(
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 973, in run_commands
    self.run_command(cmd)
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/dist.py", line 1217, in run_command
    super().run_command(command)
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
    cmd_obj.run()
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/command/test.py", line 224, in run
    self.run_tests()
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/command/test.py", line 227, in run_tests
    test = unittest.main(
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/main.py", line 124, in parseArgs
    self._do_discovery(argv[2:])
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/main.py", line 244, in _do_discovery
    self.createTests(from_discovery=True, Loader=Loader)
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/main.py", line 154, in createTests
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/loader.py", line 349, in discover
    tests = list(self._find_tests(start_dir, pattern))
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/loader.py", line 405, in _find_tests
    tests, should_recurse = self._find_test_path(
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/loader.py", line 483, in _find_test_path
    tests = self.loadTestsFromModule(package, pattern=pattern)
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/command/test.py", line 57, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/loader.py", line 191, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/command/test.py", line 57, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/loader.py", line 191, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/gnu/store/7my837qpak6w6vk8ifd99d24kmgjyp61-python-setuptools-64.0.3/lib/python3.9/site-packages/setuptools/command/test.py", line 57, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/gnu/store/mhdb6jc5ilr43mxx1zqkbwj5gj7jq6wp-python-3.9.9/lib/python3.9/unittest/loader.py", line 211, in loadTestsFromName
    raise TypeError("calling %s returned %s, not a test" %
TypeError: calling <class 'BayesicFitting.source.kernels.Uniform.Uniform'> returned Uniform: 1 if |x| < 1 else 0, not a test
```

**Comment by dokester** [comment]
Confusing.
The first line of the traceback refers to the setup.py file from BayesicFitting. All lines thereafter in the traceback are above my pay grade i.e. they refer to modules that I have nothing to say about and are beyond my responsibility.
 
It seems that it is collecting (unit)tests. For some unknown reason it selects one of the over 100 classes, called Uniform in a file kernels/Uniform.py. Subsequently it complains that Uniform is not a test... Well that is true: it is NOT. By the way the words in the last line of the traceback: "Uniform: 1 if |x| < 1 else 0" is the string representation of the Uniform class.  

In short: i don't know what I can do about this issue.

**Comment by dokester** [comment]
The real question of course is: What line in BayesicFitting causes your program to think that the Uniform class should be a test. From the traceback above I cannot decide.

**Comment by dokester** [comment]
As I don't know what to do about this, I close it. If it reappears, feel free to reopen this issue.
